jolly-jellyfish/src/page_maker/models.py

```diff
@@ -31,43 +44,48 @@ def get_user_dir_path(instance, filename):
+    name = models.CharField(max_length=80, unique=True)
     date_created = models.DateTimeField(default=timezone.now)
+    last_edit_date = models.DateTimeField(editable=False)
     author = models.ForeignKey(User, on_delete=models.CASCADE)
-    # todo: default to imgkit generated thumbnail
-    thumbnail = models.ImageField(null=True, blank=True, default=None,
-                                  upload_to='thumbnails/',
+    thumbnail = models.ImageField(null=False, blank=False,
+                                  verbose_name="thumbnail (leave empty-generated automatically)",
+                                  default='thumbnails/placeholder_img.png', upload_to='thumbnails/',
                                   validators=[validate_image_file_extension])
+    # Implausible date is just to signify it does not yet have a thumbnail
+    thumbnail_edit_date = models.DateTimeField(default=datetime(1, 1, 1), editable=False)
     template_used = models.ForeignKey(Template, on_delete=models.DO_NOTHING)
     votes = models.IntegerField(default=0)
 
     user_title = models.CharField(max_length=100, verbose_name='page title')
-    user_text_1 = models.TextField(verbose_name='subtitle paragraph')
+    user_text_1 = models.TextField(verbose_name='subtitle paragraph', blank=True)
     user_text_2 = models.TextField(verbose_name='main body paragraph')
-    user_text_3 = models.TextField(verbose_name='closing paragraph')
+    user_text_3 = models.TextField(verbose_name='closing paragraph', blank=True)
 
-    # todo: make sure background images repeat/tessellate in css if a background image is uploaded
+    # todo: support for background images with custom templates
     user_image_1 = models.ImageField(
-        null=True, blank=True, verbose_name='background image',
+        null=True, blank=True, verbose_name='header image', max_length=200,
         upload_to=get_user_dir_path, validators=[validate_image_file_extension])
     user_image_2 = models.ImageField(
-        null=True, blank=True, verbose_name='header image',
+        null=True, blank=True, verbose_name='main image', max_length=200,
         upload_to=get_user_dir_path, validators=[validate_image_file_extension])
     user_image_3 = models.ImageField(
-        null=True, blank=True, verbose_name='main image',
-        upload_to=get_user_dir_path, validators=[validate_image_file_extension])
-    user_image_4 = models.ImageField(
-        null=True, blank=True, verbose_name='footer image',
+        null=True, blank=True, verbose_name='footer image', max_length=200,
         upload_to=get_user_dir_path, validators=[validate_image_file_extension])
 
-    # todo: for tameTNT's reference: https://docs.djangoproject.com/en/3.0/topics/forms/modelforms/
-
     def was_created_recently(self):
         pass
 
     def __str__(self):
         return self.name
 
+    def save(self, *args, **kwargs):
+        """Trigger timestamp updates when object is edited"""
+        if not self.id:
```

Comment:
Could we check `self.date_created` instead of `self.id`? Just so it looks less hacky...

Reply:
This is just the standard form I belive for checking if this is an object's initial creation save call. Could be commented for clarity I guess.
```suggestion
        if not self.id:  # if initial object instantiation
```